### DISCUSSION TITLE
Store the current request in CursorPagination

### DISCUSSION THREAD

**Discussion by c-w** [encode/django-rest-framework#8761] [Ideas & Suggestions]
**Background:** Both `PageNumberPagination` and `LimitOffsetPagination` store the current request inside of the `paginate_queryset` method ([here](https://github.com/encode/django-rest-framework/blob/78cdae69997c9fd95211ec15fb4e21f4cd45e30a/rest_framework/pagination.py#L217) and [here](https://github.com/encode/django-rest-framework/blob/78cdae69997c9fd95211ec15fb4e21f4cd45e30a/rest_framework/pagination.py#L388) respectively) into the `self.request` variable. `CursorPagination` doesn't do this.

**Proposal:** I suggest a change to store the current request into the `self.request` variable also for `CursorPagination`; I'm happy to open a pull request if the proposal is accepted.

**Rationale:** Having access to `self.request` inside of the pagination classes is useful as it makes the classes easier to extend: when overriding methods we (for example) can access all the query parameters, etc. Additionally, making the access to `self.request` consistent across all three built-in pagination types makes it easier to swap around the classes while keeping any custom overridden code the same.

**Comment by lovelydinosaur** [comment]
Sure.
The consistency argument is good enough here.

**Reply by c-w** [reply]
Ok, thank you. I opened a PR: https://github.com/encode/django-rest-framework/pull/8764.
